### PR TITLE
Fix Number.isInteger(), etc. in NewTypeInference

### PR DESCRIPTION
**Pull request by Dominator008**
As suggested in #1162.

[Review on Reviewable](https://reviewable.io/reviews/google/closure-compiler/1545)

### CODE REVIEW

File: src/com/google/javascript/jscomp/NewTypeInference.java [hunk 1]
```diff
@@ -2548,12 +2548,12 @@ private JSType predicateTransformType(
       case "boolean":
       case "isBoolean":
         return booleanContext.isTrueOrTruthy()
-          ? JSType.BOOLEAN : beforeType.removeType(JSType.BOOLEAN);
+            ? JSType.BOOLEAN : beforeType.removeType(JSType.BOOLEAN);
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/NewTypeInference.java]: can you remove the extra formatting?

File: src/com/google/javascript/jscomp/CodingConventions.java [hunk 2]
```diff
@@ -304,6 +306,17 @@ public ObjectLiteralCast getObjectLiteralCast(Node callNode) {
 
     private static final long serialVersionUID = 1L;
 
+    private static final Set<String> propertyTestFunctions;
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/CodingConventions.java]: Can you switch this to use ImmutableSet.of, like this:
https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/ClosureCodingConvention.java#L345

File: test/com/google/javascript/jscomp/NewTypeInferenceES5OrLowerTest.java [hunk 3]
```diff
@@ -11064,6 +11064,48 @@ public void testIsArrayPredicate() {
         NewTypeInference.MISTYPED_ASSIGN_RHS);
   }
 
+  public void testNumberPredicates() {
```
Comment (dimvar) [inline test/com/google/javascript/jscomp/NewTypeInferenceES5OrLowerTest.java]: Let's put these in the same method as this:
https://github.com/google/closure-compiler/blob/master/test/com/google/javascript/jscomp/NewTypeInferenceES5OrLowerTest.java#L11050

File: src/com/google/javascript/jscomp/NewTypeInference.java [hunk 4]
```diff
@@ -2561,11 +2561,16 @@ private JSType predicateTransformType(
       case "number":
       case "isNumber":
         return booleanContext.isTrueOrTruthy()
-          ? JSType.NUMBER : beforeType.removeType(JSType.NUMBER);
+            ? JSType.NUMBER : beforeType.removeType(JSType.NUMBER);
+      case "isFinite":
+      case "isInteger":
+      case "isNaN":
+      case "isSafeInteger":
+        return booleanContext.isTrueOrTruthy() ? JSType.NUMBER : beforeType;
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/NewTypeInference.java]: replace this line with: return JSType.NUMBER;

**Comment by dimvar** [comment]
A high-level comment: do people actually use `Number.isFinite` as a type test? I'd expect that it should always take a number, rather than take whatever and use it as a hint to specialize to number.

**Comment by Dominator008** [comment]
It's technically correct to use these as type tests, but since it's likely an error, we can be stricter about it.

**Comment by dimvar** [comment]
Right, it seems to me also that it's an error to use these as type tests. So, if we want their arguments to be numbers already, do we need this PR?

**Comment by Dominator008** [comment]
It might still be useful:
http://closure-compiler-debugger.appspot.com/#input0%3Dfunction%2520f(arg)%2520%257B%250A%2520%2520if%2520(Number.isFinite(arg))%2520%257B%250A%2520%2520%2520%2520return%2520arg%255B0%255D%253B%250A%2520%2520%257D%2520else%2520%257B%250A%2520%2520%2520%2520return%2520arg%253B%250A%2520%2520%257D%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

Should give a warning

File: src/com/google/javascript/jscomp/NewTypeInference.java (hunk 4 above)
Comment (Dominator008) [inline src/com/google/javascript/jscomp/NewTypeInference.java]: Done.

File: test/com/google/javascript/jscomp/NewTypeInferenceES5OrLowerTest.java (hunk 3 above)
Comment (Dominator008) [inline test/com/google/javascript/jscomp/NewTypeInferenceES5OrLowerTest.java]: Done.

File: src/com/google/javascript/jscomp/CodingConventions.java (hunk 2 above)
Comment (Dominator008) [inline src/com/google/javascript/jscomp/CodingConventions.java]: Done.

File: src/com/google/javascript/jscomp/NewTypeInference.java (hunk 1 above)
Comment (Dominator008) [inline src/com/google/javascript/jscomp/NewTypeInference.java]: Done.

**Comment by dimvar** [comment]
Ah yes, this is definitely useful. Thanks Michael!

**Comment by dimvar** [comment]
LGTM. You can merge after Travis is done.

**Comment by dimvar** [comment]
Per discussion on #1563, let's abandon this PR. NTI should be able to handle type checking here without special casing for these functions.

**Comment by Dominator008** [comment]
Closing in favor of fixing NTI.

**Pull request closed without merging**